build-logic/api-extractor/build.gradle.kts

```diff
@@ -0,0 +1,19 @@
+plugins {
+    id("gradlebuild.api-extractor")
+}
+
+description = "Extract API classes from JVM classes."
+
+dependencies {
+    // TODO These should look a lot simpler.
+    //      We should probably reuse them from build-logic-commons:build-platform,
+    //      but doing so causes a lot of Kotlin tests like ManagedPropertyJavaInterOpIntegrationTest fail
+    //      with an error about a missing method:
+    //      kotlinx.metadata.jvm.KotlinClassMetadata kotlinx.metadata.jvm.KotlinClassMetadata$Companion.read(kotlinx.metadata.jvm.KotlinClassHeader)
+    implementation("org.ow2.asm:asm-tree:9.7")
+    implementation("com.google.guava:guava:32.1.2-jre") {
+        exclude("org.checkerframework", "checker-qual")
+        exclude("com.google.errorprone", "error_prone_annotations")
+        exclude("com.google.guava", "listenablefuture")
+    }
```

Comment:
:x: This needs cleaning up before merge.

Reply:
Discussed with @cobexer; this is the best we can do for now.

Reply:
Thanks, I did that. The result is still ugly, though somewhat simpler.

FTR, the original comment was about the need to specify version numbers (that are to be kept in sync with other places).